### ISSUE TITLE
Design suggestion for Installed/Updates tab

### ISSUE THREAD

**Issue by Minimalx** [Iamlooker/Droid-ify#193]
I would suggest the combination of the Installed and Updates tab, making only one tab from the two.

The Installed tab can already serve the needs of the Updates tab thus making the existence of the Updates tab kinda pointless.

By creating a new sorting order in the installed tab, replacing the current ones, the functionality of the Updates tab can be kept Inside the Installed tab.

My suggested sorting order is: applications without an available update to be sorted by name and applications with an available update to be sorted also by name but be placed top/above the applications without an available update.

For summarizing, this ui simplification would result in removing the Updates tab, removing the sorting order selector from the installed tab and using the new sorting order mentioned above in the Installed tab.

These are only my suggestions in case they are not liked this issue can be closed without reply.

**Comment by vertigo220** [comment]
I see a few problems and no real benefit with this. If you combine them, users won't be able to see as quickly how many updates there are, but rather will have to scroll to see where the sorting jumps backwards (i.e. where the apps with updates that are sorted at the top end and where the rest begin). It would also make a slightly harder to find an app, since you'd have to look through the top apps (the ones with updates) then again through the bottom ones if there is no update. And removing the sorting button/dropdown would remove the ability to use those different sorting methods which, while I rarely would need them, I do sometimes have use for seeing recently updated apps or apps I've recently installed, as I'm sure others do as well.

I admit these issues are very minor, especially the first two, and easily dealt with, but I question why functionality should be removed and issues, however minor, should be introduced, just to accomplish something I personally don't see any benefit in.

It seems to me your main goal is to simply make apps as simple/basic as possible, though I'm not sure of the reasoning behind this. I suspect it's to make it as easy to use as possible, but sometimes oversimplifying things can make things more difficult. For example, taking away sorting options can make it harder and take longer for users to find an app. If you really have issue with having three tabs and prefer only two, it seems the better solution would be to request an _additional_ sorting option called something like "Available updates" to sort the Installed tab in the way you're referring, which would then allow you to simply ignore the Updates tab, though that would really be superfluous and therefore the dev may not find it worth their time.

If the tabs are combined, I'd argue the number of updates should be put in parentheses next to the tab name and the sorting methods should not only remain, but perhaps a button should be added to toggle the pinning of the apps with updates to the top, to allow them to be sorted as they are now instead of forming two separate lists. But again, that's a fair amount of work for the dev with little to no real benefit.

**Comment by Minimalx** [comment]
If you would of tried to understand my suggestion, you would know that it does not make any harder for users to be able to see as quickly how many updates there are.
 (My english is not the best. I may phrased it very poorly. I am sorry if that is the case.)

 When you would click on the tab you would see the updates right away in abc order so you can count them.
 If you used the app you know the apps with updates have an orange mark so when you would scroll down you would know where is the end of apps with updates.

 If you would want to find an app which has an update it's in front of you in abc order the apps with updates.
 If you would want want to find an app which does not have an update you scroll down and find it, it's in abc order after the apps with updates.

 You are right in the case you don't know the app does not have an update you have to go trough the apps with updates first, however you forget 2 things.
 1. Going trough the updates first does not take significantly more time, unless you have like more than 20 apps that you did not updated yet.
 2. In that case you can use the search functionality which already present in the app...

 Anyway, it was just my suggestion. I thought I share my idea with the dev how could he make the app a bit more minimalistic/simple in case he would like.

**Comment by vertigo220** [comment]
I took a lot of time to respond to your suggestion with my own thoughts while being careful to be polite and not just disregard your opinion, because I believe everyone should have a voice and everyone's opinion counts. And in your first sentence in response, you accuse me of not even trying to understand you. I understand completely what you are suggesting, and I stand by what I said and clearly you don't understand my point. So I'll be clearer: If I don't update for a couple weeks, when I check the Updates tab, I can very quickly flick the screen to scroll and see if there are a page or two or three of updates. With your suggestion, if I were to flick the screen, it would likely scroll right past the updates into the installed section, and they would just run together. I admitted it wouldn't be a big difference, that it would be a minor issue, but, however slight, it would be a downgrade in usability which, IMO, wouldn't be worth it simply to eliminate a tab you don't want when you could instead just not use it.

As for your points:

1. No, it doesn't, neither does simply switching tabs. Also, if one has a lot of apps installed and doesn't update for even a week or two, much less longer than that, the number of updates can easily approach 20 or greater.
2. Yes, which can take longer and be more steps than simply scrolling. Adding the additional complication of having to sometimes scroll and sometimes search based on another factor (how many updates there are) makes things more complex, not less. You could use the search functionality, or you could just switch tabs.

There's a trend here: you're trying to remove parts  for simplification, and therefore functionality and, to make up for the removed functionality, you're suggesting users do things in a different, often more convoluted way. I understand your suggestion, just not the reasoning behind it.

I respect your opinion, and users should come make suggestions as that's a big part of how apps get better. I'm sure the dev will eventually look at it and make a decision one way or the other, and I was just simply offering a counter for them to consider. I'm not even completely against your idea, I just feel it needs some balance, such as _adding to_ the filter function, _not removing_ it, and putting a count of the apps with updates at the top. If you want to offer counters to me, to further explain your idea and offer more reasoning for it, or to say why you think my suggestions to balance yours aren't good, that's fine. But don't just attack me saying I didn't even try to understand your suggestions, especially after I took the time to make such a detailed response to them.

**Comment by Tanzin01** [comment]
Can we have  "downloding" apps in update section?

For example, a app is downloading and b,c,d,,e,f app are in download queue. 

we see these info in "update" page, once they are done downloading.

[sorry for weak english]

**Comment by vertigo220** [comment]
@Tanzin01 you should make that a separate issue, as it's unrelated to this one. Also, I'm not clear on what exactly you're asking for, if you me create a new issue and mention it here I can try to help you clarify it if you'd like.

**Comment by Minimalx** [comment]
@vertigo220 
It doesn't worth my time to argue with you.
I think I was clear enough to be understandable.
My suggestion was aimed to the dev not to you, in case he likes the idea, I was not trying to be forceful with it, I even wrote "in case they are not liked this issue can be closed without reply."

**Comment by Iamlooker** [comment]
Yes it is a valid point to merge bothe the pages, but the thing is the app is not architectured in a way which allows it, for now. I have this planned for future for sure
